test/rpc/test_rpc_misc.py

```diff
@@ -121,3 +150,113 @@ def test_call_with_invalid_params(params):
     # could be any legal method, just passing something to get params to fail
     ex = rpc_call(method="starknet_getClass", params=params)
     assert ex["error"] == {"code": -32602, "message": "Invalid params"}
+
+
+@pytest.mark.usefixtures("run_devnet_in_background")
+@pytest.mark.parametrize(
+    "run_devnet_in_background, input_data, expected_data",
+    [
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+            },
+            4,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "3",
+            },
+            2,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "3",
+                "to_block": "4",
+            },
+            2,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+                "chunk_size": 3,
+            },
+            3,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+                "chunk_size": 3,
+                "continuation_token": "1",
+            },
+            1,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+                "address": "0x62230eA046a9a5fbc261ac77d03c8d41e5d442db2284587570ab46455fd2488",
+            },
+            2,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+                "keys": [
+                    "0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9"
+                ],
+            },
+            2,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+                "keys": [
+                    "0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9",
+                    "0x3db3da4221c078e78bd987e54e1cc24570d89a7002cefa33e548d6c72c73f9d",
+                ],
+            },
+            4,
+        ),
+        (
+            [*PREDEPLOY_ACCOUNT_CLI_ARGS],
+            {
+                "from_block": "0",
+                "to_block": "latest",
+                "address": "0x62230eA046a9a5fbc261ac77d03c8d41e5d442db2284587570ab46455fd2488",
+                "keys": [
+                    "0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9",
+                    "0x3db3da4221c078e78bd987e54e1cc24570d89a7002cefa33e548d6c72c73f9d",
+                ],
+            },
+            2,
+        ),
+    ],
+    indirect=True,
+)
```

Comment:
To be honest I don't like this parameterized input as it makes the test file overall unclean.
It is an elegant solution but if it can be extracted into the single events_parameters config file and imported and pass like single PARAMETERS, it would be much cleaner for the test file ?

Reply:
ok, will do

Reply:
Changed